### PR TITLE
switch from manual file parsing to symfony yaml component

### LINKED ISSUE #1: switch to yml
Just a suggestion, might introduce other problems: but why not use http://symfony.com/doc/current/components/yaml/introduction.html to parse the 'txt'-file? Ok, with parsing yaml you might get more complex data structures than the simple hash map that is needed. But having a proven lib to do the parsing might have it's advantages too, when extending the file format later.

### PR DESCRIPTION
**Pull request by tolry**
closes #1 

I could also have introduced a ParserInterface and allow both implementations: manual file parsing an sf-yaml. But in this case I thought it would be unnecessary.

Since I have todos in there, this is probably [WIP]. Any thoughts on those todos or the PR in general?

### CODE REVIEW

File: src/BadgeBatch/SkillFileParser.php [hunk 1]
```diff
@@ -11,19 +13,20 @@ class SkillFileParser
...
-        foreach (explode("\n", $content) as $line) {
-            $line = trim($line);
-            if (! $line) {
-                continue;
-            }
 
-            $exp = array_filter(array_map('trim', explode(':', $line, 2)));
-            if (count($exp) != 2) {
+        // todo might throw Symfony\Component\Yaml\Exception\ParseException
+        //      do we want to catch this here or handle further up
```
Comment (larsborn) [inline src/BadgeBatch/SkillFileParser.php]: I vote for catching here and simply ignoring anything that is in invalid format.

File: src/BadgeBatch/SkillFileParser.php [hunk 2]
```diff
@@ -11,19 +13,20 @@ class SkillFileParser
...
+        $data = $parser->parse($content);
+
+        foreach ($data as $skill => $rank) {
+            // todo check for allowed values
```
Comment (larsborn) [inline src/BadgeBatch/SkillFileParser.php]: I would go for "everything is allowed" first

**Comment by larsborn** [comment]
I added a few comments. Hope I can help!

**Comment by tolry** [comment]
Thanks, I made the appropriate changes. Please check, if this fits your ideas. Should be ready to merge now.

**Pull request merged**